### PR TITLE
Use system libraries instead of embedded ones

### PR DESCRIPTION
**Pull request by gjasny**
Closes: #155

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,13 +31,54 @@ script:
   - bazel test --test_output=all //core/... //pull/...
   - bazel test --test_output=all //pull/tests/integration:scrape-test
   - bazel run -c opt //core/tests/benchmark
-  - mkdir _build
-  - cd _build
-  - cmake ..
+
+  - mkdir _build_internal_deps
+  - cd _build_internal_deps
```
Comment (jupp0r) [inline .travis.yml]: pushd here as well?

File: .travis.yml [hunk 2]
```diff
@@ -31,13 +31,54 @@ script:
   - bazel test --test_output=all //core/... //pull/...
   - bazel test --test_output=all //pull/tests/integration:scrape-test
   - bazel run -c opt //core/tests/benchmark
-  - mkdir _build
```
Comment (jupp0r) [inline .travis.yml]: should we keep building with both internal and external deps?

Comment (gjasny) [reply .travis.yml]: That makes it pretty convenient for users to embed civetweb for example. I also think it does not clutter the CMakeLists.txt files too much. For now I'd say we should keep them. Maybe flip the default to use external ones.

**Comment by gjasny** [comment]
@jupp0r I believe it's ready to be merged. I did some small cleanups like moving the 3rd party deps build for travis into a separate file. Could you please look over the PR and merge if good?

**Pull request merged**